drivers/volume/kdmp/kdmp.go

```diff
@@ -0,0 +1,300 @@
+	// driverName is the name of the aws driver implementation
+	driverName    = "kdmp"
+	proxyEndpoint = "proxy_endpoint"
+	proxyPath     = "proxy_nfs_exportpath"
+	// StorkAPIVersion current api version supported by stork
+	StorkAPIVersion = "stork.libopenstorage.org/v1alpha1"
+)
+
+type kdmp struct {
+	storkvolume.ClusterPairNotSupported
+	storkvolume.MigrationNotSupported
+	storkvolume.GroupSnapshotNotSupported
+	storkvolume.ClusterDomainsNotSupported
+	storkvolume.CloneNotSupported
+	storkvolume.SnapshotRestoreNotSupported
+}
+
+func (k *kdmp) Init(_ interface{}) error {
+	return nil
+}
+
+func (k *kdmp) String() string {
+	return driverName
+}
+
+func (k *kdmp) Stop() error {
+	return nil
+}
+
+func (k *kdmp) OwnsPVC(coreOps core.Ops, pvc *v1.PersistentVolumeClaim) bool {
+	// Try to get info from the PV since storage class could be deleted
+	pv, err := coreOps.GetPersistentVolume(pvc.Spec.VolumeName)
+	if err != nil {
+		log.PVCLog(pvc).Warnf("error getting pv %v for pvc %v: %v", pvc.Spec.VolumeName, pvc.Name, err)
+		return false
+	}
+	// for px proxy volume use kdmp by default
+	storageClassName := k8shelper.GetPersistentVolumeClaimClass(pvc)
+	if storageClassName != "" {
+		storageClass, err := storage.Instance().GetStorageClass(storageClassName)
+		if err != nil {
+			logrus.Warnf("Error getting storageclass %v for pvc %v: %v", storageClassName, pvc.Name, err)
+		}
+		if _, ok := storageClass.Parameters[proxyEndpoint]; ok {
+			return true
+		}
+	}
+	return k.OwnsPV(pv)
+}
+
+func (k *kdmp) OwnsPV(pv *v1.PersistentVolume) bool {
+
+	// for csi volumes (with or without snapshot) use kdmp driver by default
+	// except for px csi volumes
+	if pv.Spec.CSI != nil {
+		// We px csi volume fall back to default csi volume driver
+		if pv.Spec.CSI.Driver == snapv1.PortworxCsiProvisionerName ||
+			pv.Spec.CSI.Driver == snapv1.PortworxCsiDeprecatedProvisionerName {
+			return false
+		}
+		log.PVLog(pv).Tracef("CSI Owns PV: %s", pv.Name)
+		return true
+	}
+
+	return false
+}
+
+func (k *kdmp) StartBackup(backup *storkapi.ApplicationBackup,
+	pvcs []v1.PersistentVolumeClaim,
+) ([]*storkapi.ApplicationBackupVolumeInfo, error) {
+	log.ApplicationBackupLog(backup).Debugf("started kdmp backup: %v", backup.Name)
+	volumeInfos := make([]*storkapi.ApplicationBackupVolumeInfo, 0)
+	for _, pvc := range pvcs {
+		if pvc.DeletionTimestamp != nil {
+			log.ApplicationBackupLog(backup).Warnf("Ignoring PVC %v which is being deleted", pvc.Name)
+			continue
+		}
+		volumeInfo := &storkapi.ApplicationBackupVolumeInfo{}
+		volumeInfo.PersistentVolumeClaim = pvc.Name
+		volumeInfo.Namespace = pvc.Namespace
+		volumeInfo.DriverName = driverName
+		volume, err := core.Instance().GetVolumeForPersistentVolumeClaim(&pvc)
+		if err != nil {
+			return nil, fmt.Errorf("error getting volume for PVC: %v", err)
+		}
+		volumeInfo.Volume = volume
+		// create kdmp cr
+
+		dataExport := &kdmpapi.DataExport{}
+		dataExport.Name = "stork-" + pvc.Name + "-" + pvc.Namespace
```

Comment:
Was thinking instead can we the backup CR name here instead of stork that way we can actually map the backup name (given in UI) directly to the DataExport CR. Even in the pod creation, we use the DE CR name that way we have end-to-end mapping.

If we have one more backup triggered on the same PVC when the first one is in progress, it gets difficult to know which backup points to which CR.

Also good to have pvc.Namespace + pvc.Name so that we have the same uniformity referencing to the backup path (generic-backup/<ns>-<pvc name>

Reply:
good idea, changed to `backup.name-namespace-pvcname`